DataCapturing/DataCapturingService.swift

```diff
@@ -69,22 +69,19 @@ public class DataCapturingService: NSObject {
     }()
 
     /**
-     An API to store, retrieve and update captured data to the local system until the App
+     The *CoreData* stack used to store, retrieve and update captured data to the local system until the App
      can transmit it to a server.
      */
-    let persistenceLayer: PersistenceLayer
+    let coreDataStack: CoreDataManager
 
     /// An in memory storage for accelerations, before they are written to disk.
-    private var accelerationsCache = [Acceleration]()
+    var accelerationsCache = [Acceleration]()
 
     /// An in memory storage for geo locations, before they are written to disk.
-    private var locationsCache = [GeoLocation]()
+    var locationsCache = [GeoLocation]()
 
     /// The background queue used to capture data.
-    private let capturingQueue = DispatchQueue.global(qos: .userInitiated)
-
-    /// Synchronizes read and write operations on the `locationsCache` and the `accelerationsCache`.
-    private let cacheSynchronizationQueue = DispatchQueue(label: "cacheSynchronization", attributes: .concurrent)
+    let capturingQueue = DispatchQueue.global(qos: .userInitiated)
```

Comment:
Only one queue is used now. This avoids a deadlock between them both and frees some system resources.